rust_src/src/symbols.rs

```diff
@@ -154,6 +155,18 @@ impl LispSymbolRef {
     }
 }
 
+impl LispStructuralEqual for LispSymbolRef {
+    fn equal(
+        &self,
+        other: Self,
+        _equal_kind: equal_kind::Type,
+        _depth: i32,
+        _ht: &mut LispHashTableRef,
+    ) -> bool {
+        self.symbol_name().eq(other.symbol_name())
```

Comment:
@db48x is checking the variable name sufficient or should this dig deeper?

Reply:
symbols are interned so that you don't have to compare the names at all; just compare the integers